Stratis.Bitcoin.Features.Wallet/WalletFeePolicy.cs

```diff
@@ -46,42 +54,42 @@ public class WalletFeePolicy : IWalletFeePolicy
         /// </summary>
         private readonly FeeRate minRelayTxFee;
 
-        public WalletFeePolicy(BlockPolicyEstimator blockPolicyEstimator, IMempoolValidator mempoolValidator, TxMempool mempool, Network network)
+        /// <summary>
+        /// Constructs a wallet fee policy.
+        /// </summary>
+        /// <param name="blockPolicyEstimator">Block policy estimator.</param>
+        /// <param name="mempoolValidator">Memory pool validator.</param>
+        /// <param name="mempool">Memory pool.</param>
+        /// <param name="network">Network node is running on.</param>
+        /// <param name="nodeSettings">Settings for the the node.</param>
+        public WalletFeePolicy(BlockPolicyEstimator blockPolicyEstimator, IMempoolValidator mempoolValidator, TxMempool mempool, Network network, NodeSettings nodeSettings)
         {
             this.blockPolicyEstimator = blockPolicyEstimator;
             this.mempoolValidator = mempoolValidator;
             this.mempool = mempool;
 
-            this.minTxFee = new FeeRate(1000);
-            this.fallbackFee = new FeeRate(20000);
+            this.minTxFee = nodeSettings.MinTxFee;
+            this.fallbackFee = nodeSettings.FallbackTxFee;
             this.payTxFee = new FeeRate(0);
             this.maxTxFee = new Money(0.1M, MoneyUnit.BTC);
-            this.minRelayTxFee = MempoolValidator.MinRelayTxFee;
 
-            // this is a very very ugly hack
-            // the fee constants should be set at the 
-            // network level or the consensus options
-            if (network.Name.ToLower().Contains("stratis"))
-            {
-                this.minTxFee = new FeeRate(10000);
-                this.fallbackFee = new FeeRate(40000);
-                this.payTxFee = new FeeRate(0);
-                this.maxTxFee = new Money(0.1M, MoneyUnit.BTC);
-                this.minRelayTxFee = new FeeRate(10000);
-            }
+            this.minRelayTxFee = network.IsBitcoin() ? MempoolValidator.MinRelayTxFee : this.minTxFee;
```

Comment:
is the minRelayTxFee supposed to be equal to the minTxFee or do they just happen to be the same value?

Reply:
This was the existing logic. I'm not sure why it isn't using the mempool validators value the same as bitcoin. I will look into mempool validator and see if I can figure out why.

Reply:
Looks like Mempool Validator is definitely hardcoded to bitcoin a value, which explains the reason for the conditional logic in the walletfeepolicy

https://github.com/stratisproject/StratisBitcoinFullNode/blob/master/Stratis.Bitcoin.Features.MemoryPool/MempoolValidator.cs#L144

I'm not sure if this is causing other problems in mempool validator on a stratis network? Stratis mintxfee is 10000 not 1000. 

For example this check: https://github.com/stratisproject/StratisBitcoinFullNode/blob/master/Stratis.Bitcoin.Features.MemoryPool/MempoolValidator.cs#L692

Reply:
Hmm should we add MinRelayTxFee to network as well?

Reply:
Could do and then have Mempool validator use that instead of hardcoded value?